Add DVD Auto-launch thingy

poobarfoob opened the pull request:
Fix XSleep problems, set videomode, and change the printed text.

poobarfoob committed:
Update main.c

dracc commented on xboxdash/main.c:
```diff
@@ -1,4 +1,7 @@
-#include <xboxrt/debug.h>
+//#include <xboxrt/debug.h>
+//#include <windows.h>
```
This line should be uncommented.

dracc commented on xboxdash/main.c:
```diff
@@ -1,4 +1,7 @@
-#include <xboxrt/debug.h>
+//#include <xboxrt/debug.h>
```
No reason to leave this line as a comment - please remove it.

poobarfoob committed:
Update main.c

cleaning up.

poobarfoob committed:
The Third Update to main.c

hopefully this is the final one needed.

poobarfoob committed:
da fourth update to main.c

wow, this shrunk a considerable amount.

poobarfoob committed:
da fifth update to main.c

we're beginning to reach peak power here.

poobarfoob committed:
Major update to our dummy dash.

A probably very buggy and finicky dvd autolauncher has been added.

poobarfoob committed:
little adjustment

stop the screen from getting flooded with text by clearing the screen
very lazy solution but i guess it works?

poobarfoob committed:
rework some things

poobarfoob committed:
Make some more changes.

Since XLaunchXBE seems to just run in a loop, I have to add some extra text because of it.

poobarfoob committed:
Update main.c

poobarfoob committed:
rewrite and organize the code

dracc commented on xboxdash/main.c:
```diff
@@ -1,21 +1,61 @@
+    while (oldstatus != status)
+    {
+        debugClearScreen(); // clear the screen so the please insert a disc text doesnt show up here.
+        debugPrint("Tray is Open. Waiting for a disc.\n", status, oldstatus);
```
You should not send unused arguments (status and oldstatus) to debugPrint().

dracc commented on xboxdash/main.c:
```diff
@@ -1,21 +1,61 @@
+    {
+        debugClearScreen(); // clear the screen so the please insert a disc text doesnt show up here.
+        debugPrint("Tray is Open. Waiting for a disc.\n", status, oldstatus);
+        oldstatus = status; // very stupid solution do a problem that is probably very easy to fix but my pea-sized brain cant figure it out.
```
Use an `if` instead of `while`. Also, nothing stupid about setting `oldstatus` this way.

dracc commented on xboxdash/main.c:
```diff
@@ -1,21 +1,61 @@
+    HalReadSMCTrayState(&status, NULL);
+    switch (status)// using a switch now, I think this just looks cooler. if statements work fine I guess idk.
+    {
+    case 16: // if satus is 16, we know the tray is open.
```
Consider using hexadecimal instead (0x10 and 0x60), looks cleaner than regular decimal.

dracc commented on xboxdash/main.c:
```diff
@@ -1,21 +1,61 @@
+        dvdStat96();
+        Sleep(1000);
+        break;
+    }
```
Not having `default: break;` here might result in unnecessary compiler warnings.

dracc commented on xboxdash/main.c:
```diff
@@ -1,21 +1,61 @@
+{
+    debugClearScreen(); // same reasoning as above.
+    txtWall(); // disclaimer stuff if you game or application doesnt launch.
+    Sleep(5000); // give them 5 seconds to read. if it loads then nice, no need to read text.
```
Consider putting up the text after read fail instead, an extra 5 second delay every time you end up in this program won't make anyone happier.

dracc commented on xboxdash/main.c:
```diff
@@ -1,21 +1,61 @@
+    Sleep(5000); // give them 5 seconds to read. if it loads then nice, no need to read text.
+    XLaunchXBE("\\Device\\CdRom0\\default.xbe"); // launch default.xbe
+}
+void txtWall(void)
```
If you end up changing the behavior of dvdStat96, you should edit the output of this function too.
